lib/streamlit/util.py

```diff
@@ -212,3 +212,24 @@ def extract_key_query_params(
             for item in sublist
         ]
     )
+
+
+def extract_single_element_lists_query_params(
+    query_params: Dict[str, List[Any]]
```

Comment:
not a fan of these function/parameter names -- there's no reason to use/include the name `query_param`(s) because this function works for unwrapping singleton lists for dicts in general

Reply:
Totally reasonable.
How about this:
```
def extract_single_element_lists(
    d: Dict[str, List[Any]]
) -> Dict[str, Any]:
    """Extracts the single element from lists in a dictionary if the list has a length of 1.

    Parameters
    ----------
    d : dict
        The dictionary containing lists.

    Returns
    -------
    dict
        The modified dictionary with single element lists extracted.
    """
    return {
        k: v[0] if isinstance(v, list) and len(v) == 1 else v
        for k, v in d.items()
    }
```

Reply:
Seems reasonable -- maybe the word `unwrap` instead of `extract` would work better for this name. `extract` sounds like you're modifying the original dict to take those lists out rather of the dict itself rather than taking the elements out of a list

Reply:
makes sense that extract doesn't seem like the best word since we're not modifying the original dict to take those lists our. Will change it to unwrap.